File: mod/library/server/handle-message/exporter/process_seal.lua
local get_name_from_table = SMODS.load_file("mod/server/handle-message/exporter/_common/get_name_from_table.lua")()
local get_desc_from_table = SMODS.load_file("mod/server/handle-message/exporter/_common/get_desc_from_table.lua")()
local check_for_override = SMODS.load_file("mod/server/handle-message/exporter/_common/check_for_override.lua")()
local output_image = SMODS.load_file("mod/server/handle-message/exporter/_common/output_image.lua")()

local function process_seal(sets, card, seal)
	local item = {}
	output_image(seal)
	if card.ability_UIBox_table then
		item.name = get_name_from_table(card.ability_UIBox_table.name)
		item.description = get_desc_from_table(card.ability_UIBox_table.main)
	end
	item.key = seal.key
	item.set = seal.set
	if seal.mod and seal.mod.id ~= "Aura" and seal.mod.id ~= "aure_spectral" then
		item.mod = seal.mod.id
	end
	item.tags = {}
	item.image_url = check_for_override("images/" .. seal.key:gsub("?", "_") .. ".png")
	if item.name then
		sets[item.set][item.key] = item
	end
end

return process_seal

File: mod/library/server/handle-message/exporter/process-seal.ts
import {
	outputImage
} from "./_common/_exports.ts";

/**
 * Process seal data and add to sets
 *
 * @param card - The card object containing the seal
 * @param seal - The seal object to process
 */
const process_seal = (card: any, seal: typeof SMODS.Seal) => {
	const item: Record<string, any> = {};

	outputImage(seal, "seal_");

	if (card.ability_UIBox_table) {
		// Get the localized name and description from the UIBox table
		item.name = (globalThis as any).localize({
			key: card.ability_UIBox_table.name.key || seal.key,
			nodes: card.ability_UIBox_table.name.nodes || [],
			set: seal.set,
			type: "name_text"
		});

		item.description = (globalThis as any).localize({
			key: card.ability_UIBox_table.main.key || seal.key,
			nodes: card.ability_UIBox_table.main.nodes || [],
			set: seal.set,
			type: "descriptions",
			vars: card.ability_UIBox_table.main.vars || []
		});
	}

	item.key = seal.key;
	item.set = seal.set;

	if (seal.mod && seal.mod.id !== "Aura" && seal.mod.id !== "aure_spectral") {
		item.mod = seal.mod.id;
	}

	item.tags = [];
	item.image_url = `images/${(seal.key as string).replaceAll("?", "_")}.png`;

	if (item.name) {
		return item;
	}
};

export default process_seal;
